### ISSUE TITLE
Port `twisted.internet.udp` to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6003]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6003|
|Type|enhancement|
|Created|2012-09-18 15:24:27Z|
|Branch|https://github.com/twisted/twisted/tree/udp-py3-6003-2|

`twisted.internet.udp` should work on Python 3.

When doing this, udp imports in `twisted.internet.posixbase` should be moved back up to the top of the module.

Also see [#6023](https://github.com/twisted/twisted/issues/6023)

<details><summary>Searchable metadata</summary>

```
trac-id__6003 6003
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_udp_py3_6003_2 branches/udp-py3-6003-2
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347981867000000 1347981867000000
changetime__1349800598000000 1349800598000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst commented|
|-|-|

(In [35985]) Branching to 'udp-py3-6003-2'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst set owner to @exarkun|
|-|-|

http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/udp-py3-6003-2

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @itamarst|
|-|-|

  1. Intentionally not adding `twisted.internet.test.test_udp` to the ported list for now?  I guess it has yet unported dependencies.  Should we hold off on merging this until that test module can be run to make sure all is truly well?
  1. The new `test_str` in `twisted.internet.test.test_udp` is a lot like the old `test_portRepr` in `twisted.test.test_udp`.

Otherwise looks good, but the former point seems kinda significant, unless I'm missing something.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst commented|
|-|-|

`twisted.internet.test.test_udp` was ported in a previous ticket.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_6"></a>|@itamarst set status to `closed`|
|-|-|

(In [36026]) Merge udp-py3-6003-2.

Author: itamar
Review: exarkun
Fixes: [#6003](https://github.com/twisted/twisted/issues/6003)

Port twisted.internet.udp to Python 3.
